### PR TITLE
Add Python 3.8, 3.9, 3.10

### PR DESCRIPTION
**Pull request by kclowes**
## What was wrong?
Adds support for Python 3.8, 3.9, 3.10. Also updates a bunch of dependencies. 

Issue #37

## How was it fixed?

Added tests for all the added Python versions. Also:
- adds lint tests for all versions as a smoke test for dev dependency installs.
- adds odds and ends from the ethereum-python-template-repo

### To-Do

[//]: # (Stay ahead of things, add list items here!)
- [x] Clean up commit history

[//]: # (For important changes that should go into the release notes please add a newsfragment file as explained here: https://github.com/ethereum/eth-hash/blob/master/newsfragments/README.md)

[//]: # (See: https://eth-hash.readthedocs.io/en/latest/contributing.html#pull-requests)
- [x] Add entry to the [release notes](https://github.com/ethereum/eth-hash/blob/master/newsfragments/README.md)

#### Cute Animal Picture

put a cute animal picture link inside the parentheses

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -40,16 +40,18 @@ test:
...
 	rm docs/eth_hash.backends.rst
 	sphinx-apidoc -o docs/ . setup.py "*conftest*"
 	$(MAKE) -C docs clean
 	$(MAKE) -C docs html
 	$(MAKE) -C docs doctest
-	./newsfragments/validate_files.py
-	towncrier --draft --version preview
 
-docs: build-docs
+docs: validate-docs build-docs
```
Comment (fselmo) [inline Makefile:54]: Validate should be after building here:

```suggestion
docs: build-docs validate-docs
```

**Review by fselmo — CHANGES_REQUESTED** [review]
Found just one small conflict in `make docs`. I also see a pesky `towncrier` warning locally that would be nice to silence as a nice-to-have:

`.../eth-hash/docs/release_notes.rst:58: WARNING: duplicate label internal changes - for eth-hash contributors, other instance in /Users/fselmo/codigo/e/py/eth-hash/docs/release_notes.rst`

Seems like it picks up "duplicate labels" even if they are under different versions for whatever reason

**Review by pacrob — APPROVED** [review]
lgtm!

**Comment by fselmo** [comment]
> @fselmo fixed! Thanks!

ah, nice! I guess [this](https://github.com/ethereum/eth-hash/pull/40/commits/61e23557d3eeb1d7e00b473610301b8d94a79162#diff-85933aa74a2d66c3e4dcdf7a9ad8397f5a7971080d34ef1108296a7c6b69e7e3R44) is what was missing...

**Review by fselmo — APPROVED** [review]
lgtm 👌

**Comment by kclowes** [comment]
> ah, nice! I guess [this](https://github.com/ethereum/eth-hash/pull/40/commits/61e23557d3eeb1d7e00b473610301b8d94a79162#diff-85933aa74a2d66c3e4dcdf7a9ad8397f5a7971080d34ef1108296a7c6b69e7e3R44) is what was missing...

Yeah, I hope that's not too heavy-handed 🤔 I guess we'll find out

**Pull request merged**